lib/functionsEmulator.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
```

Comment:
Any meaningful changes to this file I should know about? Haven't taken as close a look at this as the rest of the PR.